### PR TITLE
smooth and by the numbers

### PR DESCRIPTION
**Pull request by Savemech**
added option ssh-keygen -N "" wich hangs asking passphrase

idk, may i was not so lazy someday and update this for centos
and even more handling openssh version handling, cause current is "something" :fearful:

### CODE REVIEW

**Comment by lpirl** [comment]
Thanks a lot @Savemech ! Will merge.

(Next time, more descriptive headlines and commit messages as well as more atomic commits would be appreciated.)

**Comment by Savemech** [comment]
@lpirl Thanks!
yea, i have hard time writing them, so i usually quote some movies :movie_camera:

File: ansible_roles/managed_host/tasks/ssh_hardening.yml [hunk 1]
```diff
@@ -142,12 +148,12 @@
   register: disable_rsa_host_key_result
   shell: "find /etc/ssh -iname '*_rsa_*' -not -name '*INSECURE' -exec rename -v 's/(.*)/$1.INSECURE/' {} \\;"
   changed_when: disable_rsa_host_key_result.stdout != ""
-  when: check_rsa_host_key_length.rc == 0
+  when: check_rsa_host_key_length.rc == 1
```
Comment (lpirl) [inline ansible_roles/managed_host/tasks/ssh_hardening.yml:151]: This is a bit confusing, in shell exit code `0` means success (i.e. _true_). I will revert those two lines.

**Comment by Savemech** [comment]
oh yea, i'm just testing so set this to 1
and kind of misunderstanding how this openssh version check & act works(this is :fearful: for me) i did disable any checks at this role, so on its just wipe out old keys and leaves only 

```
HostKey /etc/ssh/ssh_host_rsa_key
HostKey /etc/ssh/ssh_host_ed25519_key
MACs hmac-sha2-512-etm@openssh.com,hmac-sha2-256-etm@openssh.com,hmac-ripemd160-etm@openssh.com,umac-128-etm@openssh.com,hmac-sha2-512,hmac-sha2-256,hmac-ripemd160
Ciphers chacha20-poly1305@openssh.com,aes256-gcm@openssh.com,aes128-gcm@openssh.com,aes256-ctr,aes192-ctr,aes128-ctr
KexAlgorithms curve25519-sha256@libssh.org,diffie-hellman-group-exchange-sha256
```

and then regenerate host keys; 
you may also ask why, well this is "regen-keys-after-vm-clone-deploy" because usual ssh-keygen -A is just not enough; maybe someday this will go futher and i would adopt this playbook to play on other vm's like a production with rhel\centos\susel; but need another chunk of time to write distrib-specific roles(this may tricky, but rather then checking openssh version, but see at these ansible variables, and exec specific-distrib-roles
ansible -m setup ubuntu-vm012154

```
...
        "ansible_distribution": "Ubuntu", 
        "ansible_distribution_major_version": "14", 
        "ansible_distribution_release": "trusty", 
        "ansible_distribution_version": "14.04",
...
```

**Pull request merged**